File: 0118-pascals-triangle/0118-pascals-triangle.java
import java.util.List;
import java.util.ArrayList;
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> triangle = new ArrayList<>();
        triangle.add(new ArrayList<>());
        triangle.get(0).add(1);

        for(int rowNum = 1; rowNum < numRows; rowNum++){
            List<Integer> row = new ArrayList<>();
            List<Integer> preRow = triangle.get(rowNum - 1);

            row.add(1);

            for(int j = 1; j < rowNum; j++){
                row.add(preRow.get(j - 1) + preRow.get(j));
            }
            row.add(1);

            triangle.add(row);
        }
        return triangle;
    }
}

File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        if(numRows == 1) return {{1}};
        vector<vector<int>> answer = {{1}, {1, 1}};
        for(int i = 2; i < numRows; i++){
            vector<int> row(i + 1, 1);
            for(int j = 1; j < i; j++){
                    row[j] = answer[i - 1][j - 1] + answer[i - 1][j];
                }
                answer.push_back(row);
            }
        return answer;
    }
};